java/fury-core/src/main/java/org/apache/fury/BaseFury.java

```diff
@@ -55,6 +59,9 @@ public interface BaseFury {
    */
   void register(Class<?> cls, Short id, boolean createSerializer);
 
+  /** register class with given type tag which will be used for cross-language serialization. */
+  void register(Class<?> cls, String typeTag);
```

Comment:
This API is used for clang serialization, we removed it in our new xlang serialization spec in #1413 . Could we remove this API? We can add  new API when new protocol are implemented